### DISCUSSION TITLE
Labeling RDF triples

### DISCUSSION THREAD

**Discussion by matthieu-perso** [terminusdb/terminusdb#1474] [Q&A]
Hey terminusdb team, 

How could I add a label to a triple in terminus db ? My goal would be to store a url parameter to a triple  (eg [url] S -> O -> P) and to be able to search based on this label.

**Comment by GavinMendelGleason** [comment]
You can create a class Triple, that has an annotation as follows:

```json
[{  "@type": "Class",
    "@id": "Example",
    "name": "xsd:string"},
 { "@type" : "Class", 
   "@id" : "Triple",
   "subject" : "xsd:anyURI",
   "predicate" : "xsd:anyURI",
   "object" : "xsd:string",
   "annotation" : "xsd:string" }]
```
  
Queries would then look something like:

```javascript
let [s,p,o,ss,sp,so,a,l] = vars('s', 'p', 'o', 'ss', 'sp', 'so', 'a','l')
and(triple(s, "rdf:type", "@schema:Example"),
    triple(s, p, o),
    typecast(s,'xsd:string', ss), 
    typecast(p,'xsd:string', sp), 
    typecast(o,'xsd:string', so),
    triple(a, 'subject', ss), 
    triple(a, 'predicate', sp), 
    triple(a, 'object', so), 
    triple(a, 'annotation', l))
  ```

**Reply by GavinMendelGleason** [reply]
This would retrieve the annotation as the variable `l`. You could store whatever arbitrary data you'd like in the `Triple` class though, it need not only be a string.
